File: dart/protein-translation/lib/protein_translation.dart
class ProteinTranslation {
  Map<String, String> dict = {
    "AUG": "Methionine",
    "UUU": "Phenylalanine",
    "UUC": "Phenylalanine",
    "UUA": "Leucine",
    "UUG": "Leucine",
    "UCU": "Serine",
    "UCC": "Serine",
    "UCA": "Serine",
    "UCG": "Serine",
    "UAU": "Tyrosine",
    "UAC": "Tyrosine",
    "UGU": "Cysteine",
    "UGC": "Cysteine",
    "UGG": "Tryptophan",
    "UAA": "STOP",
    "UAG": "STOP",
    "UGA": "STOP",
  };
  List<String> translate(String codons) {
    List<String> lst = [];
    for (int i = 0; i < codons.length; i += 3) {
      String codon = codons.substring(i, i + 3);
      if (dict[codon] == null) {
        throw ArgumentError();
      }
      if (dict[codon] == "STOP") {
        return lst;
      }
      lst.add(dict[codon]!);
    }
    return lst;
  }
}

File: php/protein-translation/ProteinTranslation.php
<?php
declare(strict_types=1);

class ProteinTranslation
{
    public function getProteins(string $rna): array
    {
        $res = [];
        $codons = str_split($rna, 3);

        foreach ($codons as $codon) {
            $check_for_stop = match ($codon) {
                'AUG' => $res[] = 'Methionine',
                'UUU', 'UUC' => $res[] = 'Phenylalanine',
                'UUA', 'UUG' => $res[] = 'Leucine',
                'UCU', 'UCC', 'UCA', 'UCG' => $res[] = 'Serine',
                'UAU', 'UAC' => $res[] = 'Tyrosine',
                'UGU', 'UGC' => $res[] = 'Cysteine',
                'UGG' => $res[] = 'Tryptophan',
                'UAA', 'UAG', 'UGA' => 'STOP',
                default => throw new InvalidArgumentException('Invalid codon'),
            };
            if ($check_for_stop === 'STOP') {
                break;
            }
        }

        return $res;
    }
}
